patches/tock/09-add-vendor-hid-usb-interface.patch

```diff
@@ -0,0 +1,331 @@
+diff --git a/boards/nordic/nrf52840dk_opensk/Cargo.toml b/boards/nordic/nrf52840dk_opensk/Cargo.toml
+index 9e7244495..03d6f1d15 100644
+--- a/boards/nordic/nrf52840dk_opensk/Cargo.toml
++++ b/boards/nordic/nrf52840dk_opensk/Cargo.toml
+@@ -12,3 +12,6 @@ capsules = { path = "../../../capsules" }
+ kernel = { path = "../../../kernel" }
+ nrf52840 = { path = "../../../chips/nrf52840" }
+ nrf52_components = { path = "../nrf52_components" }
++
++[features]
++vendor_hid = ["capsules/vendor_hid"]
+diff --git a/capsules/Cargo.toml b/capsules/Cargo.toml
+index 680fe32b2..4f757b93d 100644
+--- a/capsules/Cargo.toml
++++ b/capsules/Cargo.toml
+@@ -8,3 +8,6 @@ edition = "2018"
+ kernel = { path = "../kernel" }
+ enum_primitive = { path = "../libraries/enum_primitive" }
+ tickv = { path = "../libraries/tickv" }
++
++[features]
++vendor_hid = []
+diff --git a/capsules/src/usb/descriptors.rs b/capsules/src/usb/descriptors.rs
+index dea2dfed9..8aac509c8 100644
+--- a/capsules/src/usb/descriptors.rs
++++ b/capsules/src/usb/descriptors.rs
+@@ -420,7 +420,7 @@ pub fn create_descriptor_buffers(
+     mut configuration_descriptor: ConfigurationDescriptor,
+     interface_descriptor: &mut [InterfaceDescriptor],
+     endpoint_descriptors: &[&[EndpointDescriptor]],
+-    hid_descriptor: Option<&HIDDescriptor>,
++    hid_descriptor: Option<&[&[&HIDDescriptor<'static>]]>,
```

Comment:
LAYERS: Missing documentation. What does the outer layer mean? (The inner layer is interface.)

Reply:
Out of date. Please ignore.